website/content/docs/agent-and-proxy/agent/exec.mdx

```diff
@@ -0,0 +1,47 @@
+---
+layout: docs
+page_title: Vault Agent Exec
+description: >-
+  Vault Agent's Exec functionality allows Vault secrets to be injected into a
+  process via environment variables using Consul Template markup.
+---
+
+# Vault Agent Exec
+
+Vault Agent's Exec functionality allows Vault secrets to be injected into a
+process via environment variables using [Consul Template markup][consul-templating-language].
+
+<Tip title="Kubernetes">
+  If you are running your applications in a Kubernetes cluster, we recommend evaluating the
+  [Vault Secrets Operator](/vault/docs/platform/k8s/vso)
+</Tip>
+
+<Warning>
+  Vault Agent Exec is in public beta. <br />
+  Please provide your feedback by opening a GitHub issue [here](https://github.com/hashicorp/vault/issues)
+</Warning>
+
+
+## Functionality
```

Comment:
I think it might be worth giving a bit of a deeper overview of interactions between `exec` and `env_template` stanzas here. It might be worth including an example with a couple env_template + exec entries as well.